Docker, core systems refused by mysql.

The docker containers in the development branch either do not work properly or the documentation is lacking.
When I run either the 4.3.0 or 4.2.0 dockers, the MySQL container refuses to communicate with core systems, only logging `bad handshake` for each tried connection.
It's the same in both the secure and insecure modes.

I tried both following the instructions by copying the `create_empty_arrowhead_db.sql` into the `docker/` directory, by running the `initSQL.sh` script, and both. The instructions mention `database initialization scripts` but only one script by name. Though this might just be the documentation lagging behind the implementation.

The service registry logs this exception
> serviceregistry    | Caused by: javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
which after googling suggests that the images might be using incompatible TLS versions https://stackoverflow.com/questions/38205947/sslhandshakeexception-no-appropriate-protocol, but if they work for you then this seems like a bug in the documentation rather than the docker files themselves.

--- a/docker/core_system_config/authorization.properties
+++ b/docker/core_system_config/authorization.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if neccessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=authorization
 spring.datasource.password=hqZFUkuHxhekio3
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/certificateauthority.properties
+++ b/docker/core_system_config/certificateauthority.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if neccessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=certificate_authority
 spring.datasource.password=FsdG6Kgf9QpPfv2
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/choreographer.properties
+++ b/docker/core_system_config/choreographer.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if neccessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=choreographer
 spring.datasource.password=Qa5yx4oBp4Y9RLX
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/datamanager.properties
+++ b/docker/core_system_config/datamanager.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if neccessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=datamanager
 spring.datasource.password=gRLjXbqu0YwYhfK
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/deviceregistry.properties
+++ b/docker/core_system_config/deviceregistry.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if necessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=device_registry
 spring.datasource.password=iooHU87hNGUalht
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/eventhandler.properties
+++ b/docker/core_system_config/eventhandler.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if neccessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=event_handler
 spring.datasource.password=gRLjXbqu9YwYhfK
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/gatekeeper.properties
+++ b/docker/core_system_config/gatekeeper.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if neccessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=gatekeeper
 spring.datasource.password=fbJKYzKhU5t8QtT
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/gateway.properties
+++ b/docker/core_system_config/gateway.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if neccessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=gateway
 spring.datasource.password=LfiSM9DpGfDEP5g
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/onboardingcontroller.properties
+++ b/docker/core_system_config/onboardingcontroller.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if necessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=onboarding_controller
 spring.datasource.password=JKgh1as5f6oi7aV
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/orchestrator.properties
+++ b/docker/core_system_config/orchestrator.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if neccessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=orchestrator
 spring.datasource.password=KbgD2mTr8DQ4vtc
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/serviceregistry.properties
+++ b/docker/core_system_config/serviceregistry.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if neccessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=service_registry
 spring.datasource.password=ZzNNpxrbZGVvfJ8
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/systemregistry.properties
+++ b/docker/core_system_config/systemregistry.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if necessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=system_registry
 spring.datasource.password=Kh12Hhgaxzo7haf
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

--- a/docker/core_system_config/translator.properties
+++ b/docker/core_system_config/translator.properties
@@ -4,7 +4,7 @@
 
 # Database connection (mandatory)
 # Change the server timezone if neccessary
-spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest
+spring.datasource.url=jdbc:mysql://mysql:3306/arrowhead?serverTimezone=Europe/Budapest&useSSL=false
 spring.datasource.username=translator
 spring.datasource.password=wozYpV58G0HUkbL
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

